utils/node_local_test.py

```diff
@@ -2322,34 +2297,33 @@ def test_complex_rename(self):
         # check fstat works.
         """
 
-        fname = os.path.join(self.dfuse.dir, 'file')
-        ofd = open(fname, 'w')
-        print(os.fstat(ofd.fileno()))
+        fname = join(self.dfuse.dir, 'file')
+        with open(fname, 'w') as ofd:
+            print(os.fstat(ofd.fileno()))
 
-        dfuse = DFuse(self.server,
-                      self.conf,
-                      pool=self.pool.id(),
-                      container=self.container,
-                      caching=False)
-        dfuse.start(v_hint='rename')
+            dfuse = DFuse(self.server,
+                          self.conf,
+                          pool=self.pool.id(),
+                          container=self.container,
+                          caching=False)
+            dfuse.start(v_hint='rename')
 
-        os.mkdir(os.path.join(dfuse.dir, 'step_dir'))
-        os.mkdir(os.path.join(dfuse.dir, 'new_dir'))
-        os.rename(os.path.join(dfuse.dir, 'file'), os.path.join(dfuse.dir, 'step_dir', 'file-new'))
+            os.mkdir(join(dfuse.dir, 'step_dir'))
+            os.mkdir(join(dfuse.dir, 'new_dir'))
+            os.rename(join(dfuse.dir, 'file'), join(dfuse.dir, 'step_dir', 'file-new'))
 
-        # This should fail, because the file has been deleted.
-        try:
-            print(os.fstat(ofd.fileno()))
-            self.fail()
-        except FileNotFoundError:
-            print('Failed to fstat() replaced file')
+            # This should fail, because the file has been deleted.
+            try:
+                print(os.fstat(ofd.fileno()))
+                self.fail()
+            except FileNotFoundError:
+                print('Failed to fstat() replaced file')
 
-        os.rename(os.path.join(self.dfuse.dir, 'step_dir', 'file-new'),
-                  os.path.join(self.dfuse.dir, 'new_dir', 'my-file'))
+                os.rename(join(self.dfuse.dir, 'step_dir', 'file-new'),
+                          join(self.dfuse.dir, 'new_dir', 'my-file'))
 
-        print(os.fstat(ofd.fileno()))
+                print(os.fstat(ofd.fileno()))
```

Comment:
Were these lines intentionally moved to inside the except?

Reply:
No, will fix tomorrow.